src/api_recievers/influxqueries.js

```diff
@@ -308,4 +308,50 @@ const grabAllPreviousStudySessionIDs = () => {
 };
 
 
-module.exports = { appData, SpecificStudySessionProcessing, grabTimesForApp, insertStudySessionData, grabAllPreviousStudySessionIDs, connectToInflux };
\ No newline at end of file
+const deleteStudySession = async (sessionId) => {
+  const deleteAPI = new DeleteAPI(influxClient);
+
+  const start = '1970-01-01T00:00:00Z'; // Start of the range (earliest possible time)
+  const stop = new Date().toISOString(); // End of the range (now)
+  const deleteKey = `_measurement="AppChange" AND QuerySession == ${sessionId}`;
+  let prevSessionDeleted = null; // Initialize outside the loop
+  
+  const headers = {
+    'Authorization': process.env.INFLUXDB_TOKEN,
+    'Content-type': 'application/json'
+  };
+
+  try {
+    await Promise.all(
+      Object.values(influxBuckets).map(async (bucket) => {
```

Comment:
I'm pretty sure map expects a return value from the callback, does this not throw an error because the callback doesn't return anything? If it does, then use forEach instead of map